src/utils/Errors.huff

```diff
@@ -0,0 +1,115 @@
+/// @title Errors
+/// @author jtriley.eth
+/// @notice Custom error utilities.
+/// The `Errors` section may be un-commented if Huff-rs supports Errors in the ABI in the future.
+/// @license Apache License, Version 2.0
+
+// https://docs.soliditylang.org/en/latest/control-structures.html?highlight=panic#panic-via-assert-and-error-via-require
+
+/* Errors */
+
+// #define error Error(string)
+// #define error Panic(uint256)
+
+/* Constants */
+
+// error Error(string);
+// Used in `require` statements
+#define constant ERROR_SIG = 0x08c379a000000000000000000000000000000000000000000000000000000000
+
+// error Panic(uint256);
+// Used in Solidity
+#define constant PANIC_SIG = 0x4e487b7100000000000000000000000000000000000000000000000000000000
+
+// Solidity Panic Codes
+#define constant COMPILER_PANIC = 0x00
+#define constant ASSERT_FALSE = 0x01
+#define constant ARITHMETIC_OVERFLOW = 0x11
+#define constant DIVIDE_BY_ZERO = 0x12
+#define constant INVALID_ENUM_VALUE = 0x21
+#define constant INVALID_STORAGE_BYTE_ARRAY = 0x22
+#define constant EMPTY_ARRAY_POP = 0x31
+#define constant ARRAY_OUT_OF_BOUNDS = 0x32
+#define constant MEMORY_TOO_LARGE = 0x41
+#define constant UNINITIALIZED_FUNCTION_POINTER = 0x51
+
+/*
+
+Solidity Require. Error `string` MUST be no greater than 32 bytes.
+
+MEMORY LAYOUT WHEN THROWN
```

Comment:
I believe strings are encoded as the pointer to the length, length, and then the left-padded contents.

ex:
```
| sig    | message length offset                                   | message length                                          | message "revert"                                        |
0x08c379a 000000000000000000000000000000000000000000000000000000020 000000000000000000000000000000000000000000000000000000006 726576657274000000000000000000000000000000000000000000000
```

See: https://docs.soliditylang.org/en/v0.8.15/control-structures.html#revert

Reply:
lol i think both of you guys are missing some zeroes

Reply:
> lol i think both of you guys are missing some zeroes

Definitely was lmao, fixed. good catch 🤝

Reply:
fixed, ty!